<launch>

    <arg name="project" default="lio_sam"/>

    <!-- Parameters -->
    <rosparam file="$(find lio_sam)/config/params.yaml" command="load" />

    <!-- Static TF: base_link -> lidar_link -->
    <node pkg="tf" type="static_transform_publisher" name="base_to_lidar" args="0 0 0 0 0 0 base_link lidar_link 10" />

    <!-- FPA Odometry Converter -->
    <node pkg="$(arg project)" type="$(arg project)_fpaOdomConverter" name="fpa_odom_converter" output="screen">
        <param name="input_topic" value="/fixposition/fpa/odometry"/>
        <param name="output_topic" value="/odometry/gps"/>
        <param name="output_frame" value="odom"/>
    </node>

    <!-- LIO-SAM Modules -->
    <node pkg="$(arg project)" type="$(arg project)_imuPreintegration"   name="$(arg project)_imuPreintegration"    output="screen" respawn="true"/>
    <node pkg="$(arg project)" type="$(arg project)_imageProjection"     name="$(arg project)_imageProjection"      output="screen" respawn="true"/>
    <node pkg="$(arg project)" type="$(arg project)_featureExtraction"   name="$(arg project)_featureExtraction"    output="screen" respawn="true"/>
    <node pkg="$(arg project)" type="$(arg project)_mapOptmization"      name="$(arg project)_mapOptmization"       output="screen" respawn="true"/>

</launch>
